portfolio/src/main/java/com/google/sps/servlets/ScheduleServlet.java

```diff
@@ -40,13 +41,32 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
 
     JSONArray eventsArray = jsonFromRequest.getJSONArray("events");
     JSONArray tasksArray = jsonFromRequest.getJSONArray("tasks");
-    String workHoursStartTimeString = (String) jsonFromRequest.get("startTime");
-    String workHoursEndTimeString = (String) jsonFromRequest.get("endTime");
+    String workHoursStartTimeString = jsonFromRequest.getString("startTime");
+    String workHoursEndTimeString = jsonFromRequest.getString("endTime");
     Instant workHoursStartTime = Instant.parse(workHoursStartTimeString);
     Instant workHoursEndTime = Instant.parse(workHoursEndTimeString);
-
+    String algorithmTypeString = jsonFromRequest.getString("algorithmType");
     Collection<CalendarEvent> events = collectEventsFromJsonArray(eventsArray);
     Collection<Task> tasks = collectTasksFromJsonArray(tasksArray);
+
+    // These are null for initialization purposes. They will either be set or
```

Comment:
I don't think the program should throw an error in case of error. The servlet can return some error response, like 400 (Bad Request). And this servlet should complete gracefully.

Reply:
Done.